tests/components/smarttub/test_init.py

```diff
@@ -11,38 +11,37 @@
 from homeassistant.setup import async_setup_component
 
 
-async def test_setup_with_no_config(hass):
+async def test_setup_with_no_config(setup_component, hass, smarttub_api):
     """Test that we do not discover anything."""
-    assert await async_setup_component(hass, smarttub.DOMAIN, {}) is True
 
     # No flows started
     assert len(hass.config_entries.flow.async_progress()) == 0
 
-    assert smarttub.const.SMARTTUB_CONTROLLER not in hass.data[smarttub.DOMAIN]
+    smarttub_api.login.assert_not_called()
 
 
-async def test_setup_entry_not_ready(hass, config_entry, smarttub_api):
+async def test_setup_entry_not_ready(setup_component, hass, config_entry, smarttub_api):
     """Test setup when the entry is not ready."""
-    assert await async_setup_component(hass, smarttub.DOMAIN, {}) is True
     smarttub_api.login.side_effect = asyncio.TimeoutError
 
     with pytest.raises(ConfigEntryNotReady):
         await smarttub.async_setup_entry(hass, config_entry)
```

Comment:
Set up the integration with `hass.config_entries.async_setup` and assert the state of the entry. In this case it will be `ENTRY_STATE_SETUP_RETRY`. Nothing will raise in the test in that case.